rubyforgood/human-essentials
# [Issue 2899] Distribution by counties!

Resolves #2899 

### Description

Ability to have show the distribution by counties. at a high level.

Addition of optional areas_served partial to partner profile.   This allows partners to select the counties they serve, and assign a percentage to each.  
Distribution by county 'report'.  Note this is not currently exportable.
Button leading to distribution by county report added to dashboard
Import of US counties
***********************
NOTE:   The Import of US counties job ( load_us_counties) will have to be run when this is promoted !!!!!!!  
***********************
Some decisions I'm not 100% sure are the best:
1/  County has full name (e.g. Washington County, Vermont)  and region (Vermont) only -- maybe it should have the short name (without region) also.  But I think we could run a job to extract it in the future if we have need.  Chose region rather than state, because we include US territories.
2/ I chose show, rather than index or something else for the action for distribution_by_county.   It is an endpoint rather than a jumping off point, though I can imagine a world where we would want to have links on each county. Will show be confusing - i.e. will people think there is a corresponding ActiveRecord?
3/  Not 100% happy with the name "served_area" for the model.   Service areas?  Served_areas at least matches up with areas_served, which is the title of the partial.  I thought about using client_share as the name,  but served_area allows the possibility of having other things about the area in the future better.

I do expect we will be asked to expand this in the future, to show (for example) diapers by county.   But let's see what happens -- that should not require more input.

Dependency:

gem "activerecord-import"   
And it does use js, including select2.

### Type of change
* New feature (non-breaking change which adds functionality)

### How Has This Been Tested?

New automatic tests
manual testing as I went along

Note that you will have to either bin/setup or just run the load_us_counties rake job in order to manually test .
